src/components/Language.js

```diff
@@ -20,13 +19,9 @@ export const Language = () => {
       const itemIndex = index - 1
       const item = items[itemIndex]
 
-      // load new language json file
-      loadMessages(item.lang).then((messages) => {
-        i18n.setLocale(item.lang)
-        i18n.load(messages)
-        setLang(i18n.locale)
-        setAppLanguage(item.lang)
-      })
+      i18n.setLocale(item.lang)
+      setLang(i18n.locale)
```

Comment:
I don't quite get why this line and the next and accessing lang in different ways.

Reply:
lang state here is only for `Language` component and allows softkey listen to lang dependency

Reply:
Hmm I don't quite get it either yet. It looks it's been this way since [the beginning of this component](https://github.com/wikimedia/wikipedia-kaios/commit/93f963caea15c89b1d3439824ea5e44d98022e1e#diff-ef8545b1621337d4615cbcf5bfb6f2dd). And what I'm seeing is that `item.lang` and `i18n.locale` are actually different when `onKeyCenter` is triggered

Reply:
Ohh sent my comment without seeing Huei's... makes sense, I see it in that dependency array of `useSoftkey`